XenAdmin/SettingsPanels/USBEditPage.cs

```diff
@@ -106,6 +106,12 @@ public void SetXenObjects(IXenObject orig, IXenObject clone)
             if (Connection == null)
                 Connection = _vm.Connection;
 
+            if (_vm != null)
+            {
+                _vm.PropertyChanged += Vm_PropertyChanged;

```

Comment:
I am not sure that we need both `UsbCollectionChanged `and `Vm_PropertyChanged` event handlers. Here we want to update the grid if an usb gets attached/detached to the VM. So I think we can register only the Vm_PropertyChanged handler and in this method we rebuild the list if the property changed is VUSBs.